packages/web/utils/rankHelpers.ts

```diff
@@ -0,0 +1,32 @@
+import { PlayerRank_Enum } from 'graphql/autogen/types';
+
+export const RANKS = [
+  PlayerRank_Enum.Diamond,
+  PlayerRank_Enum.Platinum,
+  PlayerRank_Enum.Gold,
+  PlayerRank_Enum.Silver,
+  PlayerRank_Enum.Bronze,
+];
+
+export const PLAYERS_PER_RANK = [7, 7, 7, 14, 21];
+
+// A summation of PLAYERS_PER_RANK.
+// This is the first index for which players will NOT be ranked, e.g.
+// the 56th player will be Bronze, and the 57th player will not be ranked.
+export const RANKED_CAP: number = PLAYERS_PER_RANK.reduce((sum, rankCount) => {
+  return sum + rankCount;
+}, 0);
+
+// Computes the rank for the given index. This would be the index corresponding
+// to all players ordered by total_xp DESC.
+export function computeRank(totalRankIndex: number): PlayerRank_Enum | null {
```

Comment:
you could refactor this by passing in PLAYERS_PER_RANK and RANKS to this function, and just have it return a string. I could see these two parameters being different anyway between players/patrons